src/lxc/lxccontainer.c

```diff
@@ -4659,6 +4680,152 @@ static bool do_lxcapi_restore(struct lxc_container *c, char *directory, bool ver
 
 WRAP_API_2(bool, lxcapi_restore, char *, bool)
 
+static int do_lxcapi_mount(struct lxc_container *c,
+			const char *source, const char *target,
+			const char *filesystemtype, unsigned long mountflags,
+			const void *data, struct lxc_mount *mnt) {
+	int ret = -1, len, fd, status;
+	char *template, *suff, *path;
+	pid_t pid, init_pid;
+	struct stat sb;
+
+	len = strlen(c->lxc_conf->lxc_shmount.path_host) + (sizeof("/lxcmountXXXXXX") - 1);
+	template = malloc(len);
+	if (!template) {
+		SYSERROR("Not enough memory");
+		goto out_err_free;
+	}
+	ret = snprintf(template, len, "%s/lxcmountXXXXXX", c->lxc_conf->lxc_shmount.path_host);
+	if (ret < 0 || ret >= len) {
```

Comment:
Since you will now have declared `len` as `size_t` you need to cast `ret` to `size_t` otherwise compilers will complain about comparing integers of different signs/size:

    if (ret < 0 || (size_t)ret >= len)

This cast is safe by the time you reach the right hand-side of the conjunction you know that `ret` is not negative and you know on principle grounds that `size_t` will hold a positive number at least as large as `int` can.